packages/cardano-services/src/util/TypeormProvider/TypeormProvider.ts

```diff
@@ -1,26 +1,34 @@
 import { HealthCheckResponse, Milliseconds, Provider } from '@cardano-sdk/core';
+import { InMemoryCache } from '../../InMemoryCache';
 import { TypeormService, TypeormServiceDependencies } from '../TypeormService';
 import { skip } from 'rxjs';
 
-export type TypeormProviderDependencies = Omit<TypeormServiceDependencies, 'connectionTimeout'>;
+export type TypeormProviderDependencies = Omit<TypeormServiceDependencies, 'connectionTimeout'> & {
+  healthCheckCache: InMemoryCache;
+};
 
 const unhealthy = { ok: false, reason: 'Provider error' };
 
 export abstract class TypeormProvider extends TypeormService implements Provider {
   health: HealthCheckResponse = { ok: false, reason: 'not started' };
+  healthCheckCache: InMemoryCache;
 
   constructor(name: string, dependencies: TypeormProviderDependencies) {
     super(name, { ...dependencies, connectionTimeout: Milliseconds(1000) });
     // We skip 1 to omit the initial null value of the subject
     this.dataSource$.pipe(skip(1)).subscribe((dataSource) => {
       this.health = dataSource ? { ok: true } : unhealthy;
     });
+
+    this.healthCheckCache = dependencies.healthCheckCache;
   }
 
   async healthCheck(): Promise<HealthCheckResponse> {
     if (this.state === 'running')
       try {
-        await this.withDataSource((dataSource) => dataSource.query('SELECT 1'));
+        await this.healthCheckCache.get('typeorm_db', () =>
```

Comment:
If we use the static `typeorm_db` string all the providers share the same cache entry.
We need to create a unique cache item name (maybe a combination with the `name` attribute of the `RunnableModule` ancestor class) because two distinct `TypeOrmProviders`s connect to two distinct databases.

Reply:
Done in c076aaa95833353a83284964f8b1c4e57f28d7e7